app/models/user.rb

```diff
@@ -1,9 +1,17 @@
 class User < ApplicationRecord
   has_many :articles
   has_many :bookings
+  has_many :booked_articles, through: :bookings, source: :article
   has_many :reviews
+
+  validates :username, uniqueness: true
+
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+  def today_booked_articles
+    booked_articles.where("bookings.start_date <= :date AND bookings.end_date >= :date", date: Date.today)
```

Comment:
This method renders an array of articles from the db, which are under current booking by the user. As you mentioned and explained above, we do not want a user to double-book an article, so we check on the show page whether it's currently amongst the articles booked. Happy to further explain tomorrow